lib/cw-core/src/util.rs

```diff
@@ -57,3 +57,323 @@ pub fn split_framebuffer(framebuffer: &[u8]) -> Vec<Vec<u8>> {
 
     framebuffers
 }
+
+pub fn get_different_framebuffer(previous_framebuffer: &mut [u8], framebuffer: &[u8]) -> Vec<u8> {
+    //handle incomparable frambuffer
+    if previous_framebuffer.len() != framebuffer.len() {
+        return vec![0]; //error
+    }
+
+    //handle invalid buffer
+    if previous_framebuffer.len() % 3 != 0 {
+        return vec![0];
+    }
+
+    let mut different_framebuffer_vector: Vec<u8> = vec![];
+
+    let frame_length = previous_framebuffer.len();
+    for i in (0..frame_length - 2).step_by(3) {
+        let start_i: usize = i;
+        let end_i: usize = i + 3;
+
+        //get the previous and the current rgb code
+        let prev_rgb: &[u8] = &previous_framebuffer[start_i..end_i];
+        let cur_rgb: &[u8] = &framebuffer[start_i..end_i];
+
+        if prev_rgb != cur_rgb {
+            //add index and updated rgb code to the different buffer
+            different_framebuffer_vector.extend(get_pixel_index(i));
+            different_framebuffer_vector.extend(cur_rgb);
+
+            //update the previous buffer to the latest rgb code
+            previous_framebuffer[start_i..end_i].copy_from_slice(cur_rgb);
+        }
+    }
+
+    return different_framebuffer_vector;
+}
+
+pub fn get_pixel_index(index: usize) -> [u8; 2] {
+    if index % 3 != 0 {
+        return [0u8; 2];
+    }
+
+    let index = index / 3;
+    
+    //limited in 2 bytes
+    if index > 65535 {
```

Comment:
Instead of manually typing `65525`, you can put `std::u16::MAX` there. Same output, more readable.

What would be even better is to check if the index is greater then 399, because we only have 400 pixels. When the index would be 400 or greater, there would be a mega problem in the code. 

Also with this, I would let it crash since this is not some minor problem.